src/Microsoft.SqlTools.ServiceLayer/Scripting/Contracts/ScriptingObject.cs

```diff
@@ -0,0 +1,92 @@
+﻿//
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+//
+
+using System;
+using System.Collections.Generic;
+
+namespace Microsoft.SqlTools.ServiceLayer.Scripting.Contracts
+{
+    /// <summary>
+    /// Class to represent a database object that can be scripted.
+    /// </summary>
+    public sealed class ScriptingObject : IEquatable<ScriptingObject>
+    {
+        /// <summary>
+        /// Gets or sets the database object type.  
+        /// </summary>
+        /// <remarks>
+        /// Values can be: 
+        ///     Table,
+        ///     View,
+        ///     StoredProcedure,
+        ///     UserDefinedFunction,
+        ///     UserDefinedDataType,
+        ///     User,
+        ///     Default,
+        ///     Rule,
+        ///     DatabaseRole,
+        ///     ApplicationRole,
+        ///     SqlAssembly,
+        ///     DdlTrigger,
+        ///     Synonym,
+        ///     XmlSchemaCollection,
+        ///     Schema,
+        ///     PlanGuide,
+        ///     UserDefinedType,
+        ///     UserDefinedAggregate,
+        ///     FullTextCatalog,
+        ///     UserDefinedTableType
```

Comment:
Will there be more in the future? We have a bunch of other types in the ObjectExplorer tree, can they all be added in here / should they be?